cmd/secret.go

```diff
@@ -0,0 +1,273 @@
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	sdk "github.com/ivcap-works/ivcap-cli/pkg"
+	a "github.com/ivcap-works/ivcap-cli/pkg/adapter"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+
+	"github.com/spf13/cobra"
+
+	secrettypes "github.com/ivcap-works/ivcap-core-api/http/secretmgr/types"
+)
+
+func init() {
+	rootCmd.AddCommand(secretCmd)
+
+	// LIST
+	secretCmd.AddCommand(listSecretsCmd)
+	flags := listSecretsCmd.Flags()
+	flags.StringVarP(&secPage, "page", "p", "", "page cursor")
+	flags.StringVarP(&secOffset, "offset", "", "", "offset token")
+	flags.StringVarP(&secFilter, "filter", "", "", "regexp filter by name")
+
+	// GET
+	secretCmd.AddCommand(getSecretCmd)
+
+	// SET
+	secretCmd.AddCommand(setSecretCmd)
+	flags = setSecretCmd.Flags()
+	flags.StringVarP(&secFile, "file", "f", "", "read secret from file")
+}
+
+var (
+	secPage, secOffset, secFilter, secFile string
+)
+
+var (
+	secretCmd = &cobra.Command{
+		Use:     "secret",
+		Aliases: []string{"secrets"},
+		Short:   "Set and list secrets ",
+	}
+
+	clientSkipTLSVerify *http.Client = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				// #nosec G402
+				InsecureSkipVerify: true, // Disable SSL certificate verification
+			},
+		},
+	}
+
+	listSecretsCmd = &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"l"},
+		Short:   "List existing secrets",
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			reqHost, err := getSecretHost()
+			if err != nil {
+				return err
+			}
+			var opts []a.Option
+			if strings.HasSuffix(reqHost, "ivcap.minikube") || strings.HasSuffix(reqHost, "ivcap.kube") {
+				opts = append(opts, a.WithHttpClient(clientSkipTLSVerify))
+			}
+			adpr := CreateAdapter(true, opts...)
+			req := &secrettypes.ListSecretsRequest{
+				Page:        page,
+				Limit:       limit,
+				OffsetToken: secOffset,
+				Filter:      filter,
+			}
+
+			switch outputFormat {
+			case "json", "yaml":
+				if res, err := sdk.ListSecretsRaw(context.Background(), reqHost, req, adpr, logger); err == nil {
+					return a.ReplyPrinter(res, outputFormat == "yaml")
+				} else {
+					return err
+				}
+			default:
+				if list, err := sdk.ListSecrets(context.Background(), reqHost, req, adpr, logger); err == nil {
+					printSecretsTable(list)
+				} else {
+					return err
+				}
+			}
+			return nil
+		},
+	}
+
+	getSecretCmd = &cobra.Command{
+		Use:     "get [flags] secret-name",
+		Aliases: []string{"g"},
+		Short:   "Get single secret, show its expiry time and sha1 value",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			reqHost, err := getSecretHost()
+			if err != nil {
+				return err
+			}
+			var opts []a.Option
+			if strings.HasSuffix(reqHost, "ivcap.minikube") || strings.HasSuffix(reqHost, "ivcap.kube") {
```

Comment:
You may want to add this to a "long" description and potentially add an additional flag which can override that. Not sure about the security implications, though. However, this override should somehow be made visible to users

Reply:
It will be removed, by using API gateway as ingress.